query/src/main/scala/filodb/query/exec/PromQLGrpcRemoteExec.scala

```diff
@@ -0,0 +1,88 @@
+package filodb.query.exec
+
+import java.util.concurrent.TimeUnit
+
+import io.grpc.Channel
+import io.grpc.stub.StreamObserver
+import kamon.Kamon
+import kamon.trace.Span
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.{MulticastStrategy, Observable}
+import monix.reactive.subjects.ConcurrentSubject
+
+import filodb.core.DatasetRef
+import filodb.core.query.{PromQlQueryParams, QueryContext, QuerySession}
+import filodb.core.store.ChunkSource
+import filodb.grpc._
+import filodb.grpc.GrpcMultiPartitionQueryService._
+import filodb.query.ProtoConverters._
+import filodb.query.QueryResponse
+
+
+
+trait GrpcRemoteExec extends LeafExecPlan {
+
+
+    def promQlQueryParams: PromQlQueryParams = queryContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+
+    // Since execute of overridden no one will invoke this
+    override def doExecute(source: ChunkSource, querySession: QuerySession)
+                          (implicit sched: Scheduler): ExecResult = ???
+
+    override def execute(source: ChunkSource,
```

Comment:
You're already aware and the fix is out of the scope of this PR, but I'll call it out: overridden `execute`s won't automatically apply `RangeVectorTransformers`.